### PR TITLE
Add changelog for v3.4.0

### PR DESCRIPTION
**Pull request by RaunakShah**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:

Adds changelog for upcoming `v3.4.0` release. 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: CHANGELOG/CHANGELOG-3.4.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+- Add support for cross-namespace data sources alpha feature ([#805](https://github.com/kubernetes-csi/external-provisioner/pull/805), [@ttakahashi21](https://github.com/ttakahashi21))
+- Register metrics exposed by sig-storage-lib ([#792](https://github.com/kubernetes-csi/external-provisioner/pull/792), [@RaunakShah](https://github.com/RaunakShah))
+- Update the annotation that needs to be applied to VolumeSnapshotContents from snapshot.storage.kubernetes.io/allowVolumeModeChange to snapshot.storage.kubernetes.io/allow-volume-mode-change ([#791](https://github.com/kubernetes-csi/external-provisioner/pull/791), [@RaunakShah](https://github.com/RaunakShah))
+
+### Bug or Regression
+
+- Fix string pointer comparison for source volume mode conversion ([#793](https://github.com/kubernetes-csi/external-provisioner/pull/793), [@RaunakShah](https://github.com/RaunakShah))
+- Add defence logic for PV without ClaimRef ([#796](https://github.com/kubernetes-csi/external-provisioner/pull/796), [@zezaeoh](https://github.com/zezaeoh))
+- Fix for CVE-2022-32149 ([#803](https://github.com/kubernetes-csi/external-provisioner/pull/803), [@humblec](https://github.com/humblec))
```
Comment (msau42) [inline CHANGELOG/CHANGELOG-3.4.md:19]: i'm not sure if this note is needed. From Jan's analysis, we aren't impacted: https://kubernetes.slack.com/archives/C8EJ01Z46/p1669292513960349

File: CHANGELOG/CHANGELOG-3.4.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+### Feature
+
+- Add support for cross-namespace data sources alpha feature ([#805](https://github.com/kubernetes-csi/external-provisioner/pull/805), [@ttakahashi21](https://github.com/ttakahashi21))
+- Register metrics exposed by sig-storage-lib ([#792](https://github.com/kubernetes-csi/external-provisioner/pull/792), [@RaunakShah](https://github.com/RaunakShah))
+- Update the annotation that needs to be applied to VolumeSnapshotContents from snapshot.storage.kubernetes.io/allowVolumeModeChange to snapshot.storage.kubernetes.io/allow-volume-mode-change ([#791](https://github.com/kubernetes-csi/external-provisioner/pull/791), [@RaunakShah](https://github.com/RaunakShah))
+
+### Bug or Regression
+
+- Fix string pointer comparison for source volume mode conversion ([#793](https://github.com/kubernetes-csi/external-provisioner/pull/793), [@RaunakShah](https://github.com/RaunakShah))
+- Add defence logic for PV without ClaimRef ([#796](https://github.com/kubernetes-csi/external-provisioner/pull/796), [@zezaeoh](https://github.com/zezaeoh))
```
Comment (msau42) [inline CHANGELOG/CHANGELOG-3.4.md:18]: I went ahead and updated the release notes on these PRs so that they can get automatically generated.

**Review by msau42 — COMMENTED** [review]
Sorry, https://github.com/kubernetes-csi/external-provisioner/pull/837 just merged so this will have to be regnerated.

**Comment by msau42** [comment]
Can you also update the recommended k8s verwsion to 1.26?
https://github.com/kubernetes-csi/external-provisioner/pull/837

File: CHANGELOG/CHANGELOG-3.4.md (hunk 1 above)
Comment (RaunakShah) [reply CHANGELOG/CHANGELOG-3.4.md]: Removed

File: CHANGELOG/CHANGELOG-3.4.md (hunk 2 above)
Comment (RaunakShah) [reply CHANGELOG/CHANGELOG-3.4.md]: Thanks! I re ran the script

File: README.md [hunk 3]
```diff
@@ -11,7 +11,7 @@ This information reflects the head of this branch.
 
 | Compatible with CSI Version | Container Image | [Min K8s Version](https://kubernetes-csi.github.io/docs/kubernetes-compatibility.html#minimum-version) | [Recommended K8s Version](https://kubernetes-csi.github.io/docs/kubernetes-compatibility.html#recommended-version) |
 | ------------------------------------------------------------------------------------------ | -------------------------------| --------------- | ------------- |
-| [CSI Spec v1.6.0](https://github.com/container-storage-interface/spec/releases/tag/v1.6.0) | k8s.gcr.io/sig-storage/csi-provisioner | 1.20 | 1.25 |
+| [CSI Spec v1.6.0](https://github.com/container-storage-interface/spec/releases/tag/v1.6.0) | k8s.gcr.io/sig-storage/csi-provisioner | 1.21 | 1.26 |
```
Comment (msau42) [inline README.md:14]: I think minimum version should remain at 1.20

Comment (RaunakShah) [reply README.md]: Done!

File: README.md [hunk 4]
```diff
@@ -11,7 +11,7 @@ This information reflects the head of this branch.
 
 | Compatible with CSI Version | Container Image | [Min K8s Version](https://kubernetes-csi.github.io/docs/kubernetes-compatibility.html#minimum-version) | [Recommended K8s Version](https://kubernetes-csi.github.io/docs/kubernetes-compatibility.html#recommended-version) |
 | ------------------------------------------------------------------------------------------ | -------------------------------| --------------- | ------------- |
-| [CSI Spec v1.6.0](https://github.com/container-storage-interface/spec/releases/tag/v1.6.0) | k8s.gcr.io/sig-storage/csi-provisioner | 1.21 | 1.26 |
+| [CSI Spec v1.6.0](https://github.com/container-storage-interface/spec/releases/tag/v1.6.0) | k8s.gcr.io/sig-storage/csi-provisioner | 1.20 | 1.26 |
```
Comment (humblec) [inline README.md:14]: @RaunakShah we are with 1.7 version of the spec.

**Comment by xing-yang** [comment]
@RaunakShah Can you run "release-notes" again with the latest commit?  There is an update to csi-lib-utils.

File: README.md (hunk 4 above)
Comment (RaunakShah) [reply README.md]: Updated

**Pull request merged**